File: GTP3/gtp3.cpp
#include<iostream>
#include<string.h>

using namespace std;

void imprimir_vector(int v[], int LARGO) {
        for (short i = 0; i < LARGO; i++) {
                cout << v[i] << " ";
        }
        cout << "\n\n";
}

int main() {
        struct Participante {
                char nombre[20];
                int puntajes[10];
        };

        Participante par[100], ganador;
        int cantidad, puntajes[100], indice = 0;

        cout << "Ingrese la cantidad de participantes: ";
        cin >> cantidad;

        for (int i = 0; i < cantidad; i++) {
                cout << "------ PARTICIPANTE "<<i+1<<" ------"<< endl << endl;
                
                cout << "Ingrese el nombre del participante: ";
                cin.ignore();
                cin.getline(par[i].nombre, 20, '\n');

                cout << endl;
                puntajes[i] = 0;
                for (int j = 0; j < 10; j++) {
                        cout << "       Ingrese los puntos del desafío " << j+1 << ":";
                        cin >> par[i].puntajes[j];

                        puntajes[i] += par[i].puntajes[j];
                }
        }

        for (int i = 0; i < cantidad; i++) {
                cout << "------ PARTICIPANTE "<<i+1<<" ------"<< endl << endl;
                cout << "Descripción: " << par[i].nombre << endl;
                imprimir_vector(par[i].puntajes, 10);
        }

        for (int i = 0; i < cantidad; i++) {
                if(i != 0 and puntajes[indice] < puntajes[i]) {
                        indice = i;
                }
        }

        imprimir_vector(puntajes, cantidad);
        cout << "El ganador es: " << par[indice].nombre << endl;
        
        return 0;
}

File: GTP3/gtp3.py
import random as r

NOMBRES:str = ["Forrest", "Mahmud", "Rosella", "Risa", "Katee", "Bonnibelle", "Germana", "Maximilian", "Booth", "Joly", "Luciana", "Adolph", "Freemon", "Siegfried", "Harper", "Nikolas", "Marris", "Torrance", "Viola", "Shandy", "Craggy", "Bethany", "Jewell", "Saxe", "Eldredge", "Elsa", "Gweneth", "Gerrie", "Bat", "Randall", "Trumaine", "Ed", "Stacie", "Roanna", "Rozalin", "Lexie", "Hebert", "Elinore", "Bear", "Mellisa", "Darcee", "Ari", "Carleen", "Van", "Fifine", "Minny", "Elmo", "Norina", "Adi", "Pepita", "Eloisa", "Josee", "Julita", "Gilbert", "Anatollo", "Harman", "Bill", "Waylan", "Brigham", "Kelila", "Auroora", "Mano", "Fayette", "Clo", "Rora", "Almeta", "Elwira", "Hugibert", "Jake", "Olimpia", "Pete", "Mikaela", "Ollie", "Phillipe", "Loralee", "Aggi", "Sandi", "Chelsy", "Laird", "Giselbert", "Clarissa", "Marylinda", "Leesa", "Perry", "Roxanna", "Fidelia", "Leticia", "Jenna", "Josefina", "Alli", "Robinette", "Janessa", "Adolphe", "Freddie", "Helene", "Culver", "Sylvia", "Ardith", "Rafaela", "Yetty"]

class Participante:
        nombre:str
        puntos:list[int]

        def __init__(self, nombre:str, puntos:list[int]):
                self.nombre = nombre
                self.puntos = puntos

        def puntaje(self) -> int:
                return sum(self.puntos)

        def __repr__(self) -> str:
                return f"Nombre: {self.nombre}\nPuntos: {self.puntos}"

def main():
        cantidad:int = int(input("Ingrese la cantidad de participantes: "))

        part:list[Participante] = [Participante(r.choice(NOMBRES), [r.randint(0,10) for j in range(10)]) for n in range(cantidad)]
        
        ganador:Participante = part[0]

        for i in range(1, cantidad):
                if part[i].puntaje() > ganador.puntaje():
                        ganador = part[i]

        print(part)
        print(f"El gandor es: {ganador.nombre}")
        

if __name__ == '__main__':
        main()
